### PR TITLE
#3249 clang -Wweak-vtables warning fix

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Modeler/Common/DYNSubModel.cpp [hunk 1]
```diff
@@ -1395,7 +1395,7 @@ SubModel::printInitValuesVariables(std::ofstream& fstream) {
   if (calculatedVarsInit_.size() > 0) {
     fstream << " ====== INIT CALCULATED VARIABLES VALUES ======\n";
     const vector<string>& calculatedVarNames = (*this).getCalculatedVarNamesInit();
-    for (unsigned int i = 0, iEnd = calculatedVarsInit_.size(); i < iEnd; ++i)
+    for (size_t i = 0, iEnd = calculatedVarsInit_.size(); i < iEnd; ++i)
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/Common/DYNSubModel.cpp:1398]: Not related

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestCriteria.cpp [hunk 2]
```diff
@@ -38,6 +38,7 @@
 #include "CRTCriteriaCollectionFactory.h"
 #include "TLTimelineFactory.h"
 #include <boost/algorithm/string.hpp>
+#include <boost/type_index.hpp>
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestCriteria.cpp:41]: Not related

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelTapChanger.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
+//
+// Copyright (c) 2023, RTE (http://www.rte-france.com)
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelTapChanger.cpp:2]: 2024

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Modeler/Common/DYNSubModel.cpp (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/Modeler/Common/DYNSubModel.cpp:1398]: do you mean with clang?

Comment (barondim) [reply dynawo/sources/Modeler/Common/DYNSubModel.cpp:1398]: Yes, with clang++ I have this error if I don't fix this line : 

error: implicit conversion loses integer precision: 'std::vector::size_type' (aka 'unsigned long') to 'unsigned int' [-Werror,-Wshorten-64-to-32]

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**